ContentPatcher/docs/i18n/zh/author-guide/tokens.md

```diff
@@ -0,0 +1,2011 @@
+  ```
+  注意相同键的`{{Random}}`会同步值。可设置唯一值避免：
+  ```
+  {{Random: a, b, c |key=Abigail portraits {{Time}} }}
+  ```
+
+### 固定键
+<dl>
+<dt>基础固定键：</dt>
+<dd>
+
+若需多个`Random`同步选择（如保持角色肖像与像素小人贴图一致），可指定'固定键'。相同固定键的`Random`会做出相同选择（注意列表顺序需一致）。
+
+例如，使用`abigail-outfit`固定键同步阿比盖尔的贴图和肖像：
+key:
+```js
+{
+   "Action": "Load",
+   "Target": "Characters/Abigail, Portraits/Abigail",
+   "FromFile": "assets/{{Target}}-{{Random:hood, jacket, raincoat |key=abigail-outfit}}.png"
+}
+```
+
+可在固定键中使用令牌。例如，为每个角色单独同步：
+```js
+{
+   "Action": "Load",
+   "Target": "Characters/Abigail, Portraits/Abigail, Characters/Haley, Portraits/Haley",
+   "FromFile": "assets/{{Target}}-{{Random:hood, jacket, raincoat |key={{TargetWithoutPath}}-outfit}}.png"
+}
+```
+
+<dt>高级固定键：</dt>
+<dd>
+
+固定键影响内部随机数，而非选择本身。可配合不同值（甚至不同数量）实现更有趣的功能。
+
+例如，确保阿比盖尔和海莉的服装不同：
+```js
+{
+   "Action": "Load",
+   "Target": "Characters/Abigail, Portraits/Abigail",
+   "FromFile": "assets/{{Target}}-{{Random:hood, jacket, raincoat |key=outfit}}.png"
+},
+{
+   "Action": "Load",
+   "Target": "Characters/Haley, Portraits/Haley",
+   "FromFile": "assets/{{Target}}-{{Random:jacket, raincoat, hood |key=outfit}}.png"
+}
+```
+
+</dd>
+
+<dt>固定键到底是什么？</dt>
+<dd>
+
+无固定键时，每个令牌独立选择：
+```txt
+{{Random: hood, jacket, raincoat}} = raincoat
+{{Random: hood, jacket, raincoat}} = hood
+{{Random: hood, jacket, raincoat}} = jacket
+```
+
+相同固定键时，保持同步：
+```txt
+{{Random: hood, jacket, raincoat |key=outfit}} = hood
+{{Random: hood, jacket, raincoat |key=outfit}} = hood
+{{Random: hood, jacket, raincoat |key=outfit}} = hood
+```
+
+固定键同步内部随机数，相同选项+相同键=相同值。
+
+```txt
+{{Random: hood, jacket, raincoat |key=outfit}} = 217437 modulo 3 choices = index 0 = hood
+{{Random: hood, jacket, raincoat |key=outfit}} = 217437 modulo 3 choices = index 0 = hood
+{{Random: hood, jacket, raincoat |key=outfit}} = 217437 modulo 3 choices = index 0 = hood
+```
+
+若选项顺序不同，相同索引对应不同值：
+```txt
+{{Random: hood, jacket, raincoat |key=outfit}} = 217437 modulo 3 choices = index 0 = hood
+{{Random: jacket, raincoat, hood |key=outfit}} = 217437 modulo 3 choices = index 0 = jacket
+```
+</dd>
+</dl>
+<a name="advanced"></a>
+
+## 进阶 
+<a name="query-expressions"></a>
+
+## 查询表达式 
+_查询表达式_ 是一组可计算为数字，`true`/`false`或文本的算术和逻辑表达式。
+
+### 用法
+使用`Query`执行查询表达式。可用作占位符或条件，可包含嵌套令牌：
+```js
+{
+   "Format": "2.7.0",
+   "Changes": [
+      {
+         "Action": "EditData",
+         "Target": "Characters/Dialogue/Abigail",
+         "Entries": {
+            "Mon": "Hard to imagine you only arrived {{query: {{DaysPlayed}} / 28}} months ago!"
+         },
+         "When": {
+            "query: {{Hearts:Abigail}} >= 10": true
+         }
+      }
+   ]
+}
+```
+
+文本值需用单引号包裹（包括返回文本的令牌）：
+```js
+"Query: '{{Season}}' = 'spring'": true
+```
+
+表达式不区分大小写，包括文本比较。
+
+### 注意事项
+查询表达式功能强大，但需注意：
+
+* 查询表达式 **很难验证** 无效表达式通常不会预先警告，仅在应用补丁时失败。需仔细测试表达式的功能，可以用[`patch parse`](troubleshooting.md#parse)检查新表达式。
+* 查询表达式 **可能错误读取文本** 例如，玩家名含单引号`D'Artagnan`，时，以下表达式会因语法错误失败：
```

Comment:
> 查询表达式 **可能错误读取文本** 例如，玩家名含单引号`D'Artagnan`，时，以下表达式会因语法错误失败：

I think this changes the meaning a bit? The original text stresses that the `Query` evaluates the final result after tokens are parsed, and gives `D'Artagnan` as an example of how that can lead to errors. The translation seems to just say that `Query` can read text incorrectly.